papis/commands/mv.py

```diff
@@ -55,58 +105,92 @@ def run(document: papis.document.Document,
-
     completer = prompt_toolkit.completion.PathCompleter(
         only_directories=True,
-        get_paths=lambda: [lib_dir]
+        get_paths=lambda: [str(lib_dir)]
     )
 
-    try:
-        new_folder = os.path.join(
-            lib_dir,
-            prompt_toolkit.prompt(
-                message=(
-                    "Enter directory  : (Tab completion enabled)\n"
-                    "Current directory: ({dir})\n".format(
-                        dir=document.get_main_folder_name()
-                    ) + ">  "
-                ),
+    # If the user hasn't passed ``--folder_name``, we'll prompt for one that
+    # will be used throughout the process.
+    dest_path = None
+    if folder_name is None:
+        try:
+            subfolder = prompt_toolkit.prompt(
+                message="Destination folder: (Tab completion enabled)\n> ",
                 completer=completer,
-                complete_while_typing=True
-            ))
-    except Exception as exc:
-        logger.error("Failed to choose directory.", exc_info=exc)
-        return
-
-    logger.info("New document folder: '%s'.", new_folder)
-
-    if not os.path.exists(new_folder):
-        logger.info("Creating path '%s'.", new_folder)
-        os.makedirs(new_folder, mode=papis.config.getint("dir-umask") or 0o666)
-
-    run(document, new_folder, git=git)
+                complete_while_typing=True)
+            dest_path = os.path.join(lib_dir, subfolder)
+        except Exception as exc:
+            logger.error("Failed to choose directory.", exc_info=exc)
+            return
+
+    if not git and os.path.exists(os.path.join(lib_dir, ".git")):
+        git = papis.tui.utils.confirm("Add the changes to git?")
+
+    moves = []
+    for document in documents:
+        # If the user has passed ``--folder-name``, use that to generate the subfolder
+        if folder_name is not None:
+            subfolder = papis.format.format(folder_name, document)
+            dest_path = os.path.join(lib_dir, subfolder)
+
+        # This should never be reached, but just in case.
+        if dest_path is None:
+            logger.error("Failed to construct a folder path for the document.")
+            return
+
+        # Check against erroneous documents
+        doc_main_folder = document.get_main_folder()
+        if not doc_main_folder:
+            raise DocumentFolderNotFound(papis.document.describe(document))
+
+        # This can happen when using format but a key doesn't exist and so the
+        # format expression evaluates to ''.
+        if dest_path.rstrip(os.path.sep) == os.path.dirname(doc_main_folder):
+            logger.warning("Skipping '%s': its source and destination are the same",
+                           doc_main_folder)
+            continue
+
+        if not os.path.exists(dest_path):
+            logger.info("Creating path '%s'.", dest_path)
+            os.makedirs(dest_path, mode=papis.config.getint("dir-umask") or 0o666)
```

Comment:
Does this correctly check that we're not overwritting existing documents? I didn't see any checks in `run()` either.

Reply:
I don't think it does! It should check and skip the move, like `rename` (IIRC) does.